homeassistant/components/subaru/sensor.py

```diff
@@ -0,0 +1,251 @@
+"""Support for the Subaru sensors."""
+import logging
+
+import subarulink.const as sc
+
+from homeassistant.const import (
+    LENGTH_KILOMETERS,
+    LENGTH_MILES,
+    PRESSURE_HPA,
+    TEMP_CELSIUS,
+    TIME_MINUTES,
+    UNIT_PERCENTAGE,
+    VOLT,
+    VOLUME_GALLONS,
+    VOLUME_LITERS,
+)
+from homeassistant.util.distance import convert as dist_convert
+from homeassistant.util.unit_system import (
+    IMPERIAL_SYSTEM,
+    LENGTH_UNITS,
+    PRESSURE_UNITS,
+    TEMPERATURE_UNITS,
+)
+from homeassistant.util.volume import convert as vol_convert
+
+from . import DOMAIN as SUBARU_DOMAIN, SubaruEntity
+
+_LOGGER = logging.getLogger(__name__)
+L_PER_GAL = vol_convert(1, VOLUME_GALLONS, VOLUME_LITERS)
+KM_PER_MI = dist_convert(1, LENGTH_MILES, LENGTH_KILOMETERS)
+
+# Fuel Economy Constants
+FUEL_CONSUMPTION_L_PER_100KM = "L/100km"
+FUEL_CONSUMPTION_MPG = "mi/gal"
+FUEL_CONSUMPTION_UNITS = [FUEL_CONSUMPTION_L_PER_100KM, FUEL_CONSUMPTION_MPG]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Subaru sensors by config_entry."""
+    coordinator = hass.data[SUBARU_DOMAIN][config_entry.entry_id]["coordinator"]
+    vehicle_info = hass.data[SUBARU_DOMAIN][config_entry.entry_id]["vehicles"]
+    entities = []
+    for vin in vehicle_info.keys():
+        _create_sensor_entities(entities, vehicle_info[vin], coordinator, hass)
+    async_add_entities(entities, True)
+
+
+def _create_sensor_entities(entities, vehicle_info, coordinator, hass):
+    if vehicle_info["has_safety"]:
+        entities.append(
```

Comment:
Maybe have a list of sensors and iterate over it?

Reply:
Refactored in 934d31e to use a list.